File: Day-4/labs/java-kafka-consumer-lab/src/main/java/com/kafka/consumer/lab/AutoCommitConsumer.java
package com.kafka.consumer.lab;

import java.time.Duration;
import java.util.Collections;
import java.util.Properties;
import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.clients.consumer.ConsumerRecords;
import org.apache.kafka.clients.consumer.KafkaConsumer;

/** Lab 03 - enable.auto.commit=true (default interval 5s). */
public final class AutoCommitConsumer {

    public static void main(String[] args) {
        String groupId = ConsumerConfigFactory.arg(args, 0, "commit-lab-group");
        Properties props = ConsumerConfigFactory.baseProps(ConsumerConfigFactory.DEFAULT_BOOTSTRAP, groupId);
        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
        props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "5000");

        try (KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props)) {
            consumer.subscribe(Collections.singletonList(ConsumerConfigFactory.DEMO_TOPIC));
            System.out.println("AutoCommitConsumer: enable.auto.commit=true, interval=5000ms");

            while (true) {
                ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(500));
                for (ConsumerRecord<String, String> record : records) {
                    System.out.printf("AUTO partition=%d offset=%d value=%s%n",
                            record.partition(), record.offset(), record.value());
                }
            }
        }
    }
}

File: Day-4/labs/python-kafka-consumer-lab/auto_commit_consumer.py
"""Lab 03 - auto commit enabled."""
import sys
from kafka import KafkaConsumer
from config import BOOTSTRAP_SERVERS, DEMO_TOPIC


def main():
    group_id = sys.argv[1] if len(sys.argv) > 1 else "commit-lab-group"
    consumer = KafkaConsumer(
        DEMO_TOPIC,
        bootstrap_servers=BOOTSTRAP_SERVERS,
        group_id=group_id,
        auto_offset_reset="earliest",
        enable_auto_commit=True,
        auto_commit_interval_ms=5000,
        value_deserializer=lambda v: v.decode("utf-8"),
    )
    print("auto_commit_consumer: interval=5000ms")
    for msg in consumer:
        print(f"AUTO P={msg.partition} offset={msg.offset} value={msg.value}")


if __name__ == "__main__":
    main()
